pkg/xds/server/callbacks/dataplane_sync_tracker.go

```diff
@@ -89,18 +89,16 @@ func (t *dataplaneSyncTracker) OnStreamRequest(streamID core_xds.StreamID, req u
 		defer t.Unlock()
 		streams := t.dpStreams[dataplaneKey]
 		if streams.activeStreams == nil {
-			streams.activeStreams = map[core_xds.StreamID]bool{}
+			streams.activeStreams = map[core_xds.StreamID]context.CancelFunc{}
 		}
-		streams.activeStreams[streamID] = true
-		if streams.watchdogCancel == nil { // watchdog was not started yet
-			stopCh := make(chan struct{})
-			streams.watchdogCancel = func() {
-				close(stopCh)
-			}
-			// kick off watchdog for that Dataplane
-			go t.newDataplaneWatchdog(proxyId, streamID).Start(stopCh)
-			dataplaneSyncTrackerLog.V(1).Info("started Watchdog for a Dataplane", "streamid", streamID, "proxyId", proxyId, "dataplaneKey", dataplaneKey)
+		stopCh := make(chan struct{})
```

Comment:
Are you saying that it's enough to depend on `streamsAssociation` to know that we haven't already started the watchdog for this stream?